src/ResourceManager/DataFactories/Commands.DataFactories/Gateway/GetAzureDataFactoryGatewayAuthKeyCommand.cs

```diff
@@ -0,0 +1,60 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.DataFactories.Models;
+using Microsoft.Azure.Commands.DataFactories.Properties;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation;
+using System.Security.Permissions;
+
+namespace Microsoft.Azure.Commands.DataFactories
+{
+    [Cmdlet(VerbsCommon.Get, Constants.GatewayAuthKey, DefaultParameterSetName = ByFactoryName), OutputType(typeof(PSDataFactoryGatewayAuthKey))]
+    public class GetAzureDataFactoryGatewayAuthKeyCommand : DataFactoryBaseCmdlet
+    {
+        [Parameter(ParameterSetName = ByFactoryObject, Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true,
```

Comment:
@nicolaszhang the scenario you mentioned is definitely valid, but using `ValueFromPipelineByPropertyName` here doesn't make too much sense unless a user would be piping an object with **a property whose name is DataFactory**. If you switch this to `ValueFromPipeline` then it allows a user to pipe the resulting `PSDataFactory` object from a cmdlet (such as `Get-AzureRmDataFactory`) to this cmdlet. It looks like the cmdlet you mentioned that previously used this pattern was incorrect.

An additional way of doing what you want is to create a separate parameter set that contains parameters `Name` (alias to `DataFactoryName`) and `ResourceGroupName`, and a user can provide those instead of the entire `PSDataFactory` object (you can also give these parameters the `ValueFromPipelineByPropertyName` attribute)